core/org.eclipse.cdt.core.tests/model/org/eclipse/cdt/core/build/DefaultBuildConfigurationNameTests.java

```diff
@@ -0,0 +1,438 @@
+		// Get the default build config from the project (it always has one)
+		IBuildConfiguration buildConfig = project.getBuildConfig(IBuildConfiguration.DEFAULT_CONFIG_NAME);
+		ICBuildConfiguration cBuildConfig = provider.getCBuildConfiguration(buildConfig, "default");
+		assertThat(cBuildConfig, is(nullValue()));
+	}
+
+	/**
+	 * Tests that Autotools build configurations are never created with the default name.
+	 *
+	 * @see "Test summary:" {@link #cMakeBuildConfigurationProviderGetCBuildConfiguration()}
+	 */
+	@Test
+	public void autotoolsBuildConfigurationProviderGetCBuildConfiguration() throws Exception {
+		ICBuildConfigurationProvider provider = new AutotoolsBuildConfigurationProvider();
+		// Get the default build config from the project (it always has one)
+		IBuildConfiguration buildConfig = project.getBuildConfig(IBuildConfiguration.DEFAULT_CONFIG_NAME);
+		ICBuildConfiguration cBuildConfig = provider.getCBuildConfiguration(buildConfig, "default");
+		assertThat(cBuildConfig, is(nullValue()));
+	}
+
+	// Test infrastructure...
+
+	private void waitForLaunchBarTracker() throws OperationCanceledException, InterruptedException {
+		Job.getJobManager().join(CoreBuildLaunchBarTracker.JOB_FAMILY_CORE_BUILD_LAUNCH_BAR_TRACKER, null);
+	}
+
+	private IProject createCMakeProject() throws Exception {
+		// Create a  plain Eclipse project
+		IProject project = ResourceHelper.createProject(this.getName());
+		// Add C/C++ and CMake natures to make it a CMake project
+		IProjectDescription description = project.getDescription();
+		description.setNatureIds(
+				new String[] { CProjectNature.C_NATURE_ID, CCProjectNature.CC_NATURE_ID, CMakeNature.ID });
+		project.setDescription(description, null);
+		return project;
+	}
+
+	/**
+	 * Add a mocked toolchain, which acts like a gcc toolchain and has Platform related OS ans ARCH.
+	 */
+	private void addMockToolchain() throws CoreException {
+		// Setup a toolchain ready to use for creating the ICBuildConfiguration
+		mockToolchain = mock(IToolChain.class);
+		when(mockToolchain.getProperty(anyString())).thenAnswer(new Answer<>() {
+			@Override
+			public Object answer(InvocationOnMock invocation) throws Throwable {
+				String key = invocation.getArgument(0);
+				return switch (key) {
+				case IToolChain.ATTR_OS -> Platform.getOS();
+				case IToolChain.ATTR_ARCH -> Platform.getOSArch();
+				case "name" -> null; // name=Local
+				default -> null;
+				};
+			}
+		});
+		when(mockToolchain.getTypeId()).thenReturn("tc_typeId");
+		when(mockToolchain.getId()).thenReturn("tcId");
+		// I don't understand why it is necessary to mock IToolChain.matches ???
+		when(mockToolchain.matches(anyMap())).thenAnswer(new Answer<>() {
+			@Override
+			public Object answer(InvocationOnMock invocation) throws Throwable {
+				Map<String, String> properties = invocation.getArgument(0);
+				// Copied from IToolChain.matches(Map<String, String>)
+				/*
+				 * Is this logic correct? With values
```

Comment:
I don't know the answer - the IToolChain version has this comment "// If toolchain doesn't have this property, it doesn't care" - so it seems on purpose

It was written this way because of https://bugs.eclipse.org/bugs/show_bug.cgi?id=548980